programs/sbf/tests/programs.rs

```diff
@@ -1232,6 +1235,69 @@ fn test_program_sbf_caller_has_access_to_cpi_program() {
         ..
     } = create_genesis_config(50);
     let bank = Bank::new_for_tests(&genesis_config);
+
+    let bank = Arc::new(bank);
+    let mut bank_client = BankClient::new_shared(bank.clone());
+
+    let caller_pubkey = load_program(
+        &bank_client,
+        &bpf_loader::id(),
+        &mint_keypair,
+        "solana_sbf_rust_caller_access",
+    );
+    let (_, caller2_pubkey) = load_program_and_advance_slot(
+        &mut bank_client,
+        &bpf_loader::id(),
+        &mint_keypair,
+        "solana_sbf_rust_caller_access",
+    );
+
+    // Here we are not passing caller2_pubkey down, but cpi should still succeed
+    let account_metas = vec![
+        AccountMeta::new_readonly(caller_pubkey, false),
+        AccountMeta::new_readonly(caller2_pubkey, false),
```

Comment:
But this is passing `caller2_pubkey` down right here.

Reply:
Yes, that was not great. I've created some better tests below